src/synthesizer/load_data/load_bluetides.py

```diff
@@ -0,0 +1,320 @@
+            )
+
+        self.accretion_unit_conversion = 1e10 / (
+            self.kpc / 1e3 / self.yr
+        )  # to solar masses per year
+        self.acrtolum = (
+            self.eta * self.c * self.c * (self.m_solar / self.yr) * 10**7
+        )  # Accounts for unit conversion to physical units already happening
+        self.z = z
+
+        # Load all BlueTides data
+        if self.z == 7:
+            sunset = BigFile(bluetides_data_folder + "sunset_208")
+            pig_zoo = [bluetides_data_folder + "PIG_208"]
+            length_of_bhar = end_of_arr
+        elif self.z == 6.5:
+            sunset = BigFile(bluetides_data_folder + "sunset_271")
+            pig_zoo = [bluetides_data_folder + "PIG_271"]
+            length_of_bhar = end_of_arr
+        elif self.z == 7.5:
+            pig_zoo = [bluetides_data_folder + "PIG_129"]
+            sunset = BigFile(bluetides_data_folder + "sunset_129")
+            length_of_bhar = end_of_arr
+        elif self.z == 8:
+            pig_zoo = [bluetides_data_folder + "PIG_086"]
+            sunset = BigFile(bluetides_data_folder + "sunset_086")
+            length_of_bhar = end_of_arr
+        else:
+            print("z value not supported")
+            exit()
+
+        # Determine & store galaxy properties
+        file = pig_zoo[0]  # filename
+        pig = BigFile(file)  # load data
+
+        print(f"Chosen redshift = {self.z}")
+        self.z = self.redshift = float(
+            1.0 / pig.open("Header").attrs["Time"] - 1
+        )
+        print(f"Actual bluetides redshift = {self.z}")
+
+        # If there's no OffsetByType field, use below to get the same output:
+        # get FOF info to know which stars correspond to which galaxies
+        length_by_type = pig.open("FOFGroups/LengthByType")[0:length_of_bhar]
+        offset_by_type = np.cumsum(length_by_type, axis=0)
+        a1 = np.zeros((1, 6)).astype(int)
+        offset_by_type = np.append(a1, offset_by_type, axis=0).astype(int)
+        offset_by_type = np.transpose(offset_by_type)
+
+        self.bh_offset = np.transpose(offset_by_type[5])
+        self.StarOffset = StarOffset = np.transpose(offset_by_type[4])
+
+        a_cur = 1.0 / (1.0 + self.redshift)
+        sft_space = np.linspace(0, a_cur, 100)
+        mpc_to_km = 3.086e19
+        sec_to_yr = 3.17098e-8
+        age_space = [
+            get_age(SFT, 1.0 / (1.0 + self.redshift)) * mpc_to_km * sec_to_yr
+            for SFT in sft_space
+        ]  # yr
+        self.gen_SFT_to_age = interpolate.interp1d(
+            sft_space, age_space, fill_value="extrapolate"
+        )  # get an interpolator to translate
+
+        self.bhar = (
+            pig.open("5/BlackholeAccretionRate")[0:length_of_bhar]
+            * self.accretion_unit_conversion
+        )  # solar masses per year
+
+        # Indexes of BH with largest accretion rate,
+        # everything is sorted by this array
+        self.luminous_BH_indices = np.argsort(-self.bhar)
```

Comment:
The only reason I sort is just because I'm most interested in the largest BHARs, but I realize that view is very biased. BlueTides data is not sorted except very roughly by number of star particles in a halo so this would ensure some sorting is done. Let me know if you'd prefer me to take it out.

Reply:
I think it's fine for now, we may want to sort by something like mass in future for consistency with other load methods

Reply:
Sorting has been turned into conditional statement.